### ISSUE TITLE
bug: placeholder templater silently suppresses lint violations for rules with broad anchors

### ISSUE THREAD

**Issue by ben-duivenvoorden** [sqlfluff/sqlfluff#7817]
## Summary

When using the `placeholder` templater, lint rules with broad anchors (e.g. statement-level `SegmentSeekerCrawler`) silently produce **zero violations** for SQL that should fail. The same SQL linted with literal values correctly reports violations.

## Expected Behaviour

`sqlfluff lint` should report the same violations regardless of whether a value is written as a literal or substituted via a placeholder template variable.

## Observed Behaviour

Rules such as AM04 (ambiguous column count in `UNION`), AM07 (`UNION` vs `UNION ALL`), and ST11 (set operation without brackets) return **no violations** when the SQL contains placeholder substitutions like `{{schema}}`, even though the substituted values have no effect on the rule logic.

## Steps to Reproduce

```bash
# Create test SQL and config
echo "SELECT a, b FROM {{schema}}.t1 UNION SELECT c FROM {{schema}}.t2" > /tmp/test.sql
echo -e "[sqlfluff]\ndialect = ansi\ntemplater = placeholder\n[sqlfluff:templater:placeholder]\nparam_regex = \$\{[a-zA-Z_]+\}\nparam_style = dollar" > /tmp/.sqlfluff

# Lint - expect AM04/AM07 violations but get NONE
sqlfluff lint /tmp/test.sql --config /tmp/.sqlfluff
```

Compare with literal SQL (no placeholders):
```bash
echo "SELECT a, b FROM myschema.t1 UNION SELECT c FROM myschema.t2" > /tmp/test_literal.sql
sqlfluff lint /tmp/test_literal.sql --dialect ansi
# Correctly reports AM04, AM07 violations
```

## Root Cause

`Linter.remove_templated_errors()` discards any violation whose anchor position marker returns `is_literal() == False`. The placeholder templater creates `RawFileSlice` objects with `slice_type="templated"` for substitutions, causing `TemplatedFile.is_source_slice_literal()` to return `False` for any source span overlapping a placeholder. Rules with broad anchors (spanning an entire statement) inevitably overlap placeholder regions and get silently dropped.

## Affected Rules

Any rule using `SegmentSeekerCrawler` with broad segment types is affected. Confirmed affected rules include:

| Rule | Description | Confirmed |
|------|-------------|-----------|
| AL04 | Unique column aliases | Yes |
| AL05 | Unused CTE aliases | Yes |
| AL06 | Short alias length | Yes |
| AL07 | Avoid right joins | Yes |
| AL08 | Distinct used with parentheses | Yes |
| AL09 | Self-aliasing | Yes |
| AM04 | Ambiguous column count in set operations | Yes |
| AM07 | UNION vs UNION ALL | Yes |
| AM09 | ORDER BY in set operations | Yes |
| CV03 | Trailing commas | Yes |
| LT09 | SELECT targets on separate lines | Yes |
| LT10 | Inconsistent indent of SELECT targets | Yes |
| ST11 | Set operations without brackets | Yes |

## Dialect

All dialects using the placeholder templater are affected.

## Version

Tested against `main` branch (commit 39b5a60).
